src/Core/src/Hosting/MauiApp.cs

```diff
@@ -32,12 +33,33 @@ internal MauiApp(IServiceProvider services)
 
 		/// <inheritdoc />
 		public void Dispose()
+		{
+			DisposeConfiguration();
+
+			(_services as IDisposable)?.Dispose();
+		}
+
+		/// <inheritdoc />
+		public async ValueTask DisposeAsync()
+		{
+			DisposeConfiguration();
+
+			if (_services is IAsyncDisposable asyncDisposable)
+			{
+				// Fire and forget because this is called from a sync context
+				await asyncDisposable.DisposeAsync();
```

Comment:
@PureWeen should this be using the FireAndForget method to avoid breaking the app? Maybe we can't log but we should at least catch the exceptions?